data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/opponent_weapon_has_no_special.cfg

```diff
@@ -89,3 +89,89 @@
         {SUCCEED}
     [/event]
 )}
+
+#####
+# API(s) being tested: [attacks][filter_opponent][filter_weapon][not]special_id_active=
+##
+# Actions:
+# Give bob an attacks ability which is active when the opponent don't has an ability with a specific id.
+# remove all attack to alice.
+# Have bob and alice fight.
+##
+# Expected end state:
+# bob's ability does activate, leaving him with 3 strikes.
+#####
+{GENERIC_UNIT_TEST "opponent_weapon_has_no_weapon" (
+    [event]
+        name=start
+
+        [object]
+            silent=yes
+            [effect]
+                apply_to=new_ability
+                [abilities]
+                    [attacks]
+                        value=3
+                        [filter_opponent]
+                            race=elf
+                            [filter_weapon]
+                                [not]
+                                    special_id_active=test_cth
+                                [/not]
+                            [/filter_weapon]
+                        [/filter_opponent]
+                    [/attacks]
+                    [damage]
+                        value=1
+                    [/damage]
+                    [chance_to_hit]
+                        value=100
+                    [/chance_to_hit]
+                [/abilities]
+            [/effect]
+            [filter]
+                id=bob
+            [/filter]
+        [/object]
+
+        [object]
+            silent=yes
+            [effect]
+                apply_to=remove_attacks
+            [/effect]
+            [filter]
+                id=alice
+            [/filter]
+        [/object]
+
+        [do_command]
+            [move]
+                x=7,12
+                y=3,3
```

Comment:
This should be a COMMON_KEEP test. While there's a lot of tests with the old code, this should be based on the modernised tests instead of the old ones.

Reply:
fixed